processor.go

```diff
@@ -80,36 +83,45 @@ func (pro *Processor) OnProposal(proposal *message.Proposal) error {
 	// 1) try to confirm the parent vertex of the proposal
 	err := pro.confirmParent(proposal)
 	if err != nil {
+		// this error contains information about the reason for non-confirmation and is eventually handled in the caller scope
```

Comment:
Error handling is not really relevant here. The point is, what does confirmation of a parent mean?

Reply:
Confirmation of a parent tries to confirm the parent vertex ID of the candidate and to remove its records from peer own buffer/cache.
This is done in order to avoid some future candidate (faulty) pointing to the same parent to be considered its descendant.
In this way the peer is first confirming the parent ID via `pro.graph.Confirm` method, and next time it confirms a parent that is not present in its cache - it will know candidate is a doubling one because the parent records were already removed its ancestry record.